File: src/tokenjar/filters/ansi.py
from __future__ import annotations

import re

# Regex that matches ALL ANSI escape sequences (colors, cursor, etc.)
ANSI_PATTERN = re.compile(r"\x1b\[[0-9;]*[a-zA-Z]|\x1b\][^\x07]*\x07|\x1b\([B0UK]|\r")


def strip_ansi(text: str) -> str:
    """Remove all ANSI escape codes from text."""
    return ANSI_PATTERN.sub("", text)

File: crates/tokenjar-core/src/filters/ansi.rs
//! Ultra-fast ANSI terminal escape sequence stripper.

use regex::Regex;
use std::sync::LazyLock;

static ANSI_REGEX: LazyLock<Regex> =
    LazyLock::new(|| Regex::new(r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])").unwrap());

/// Strip all ANSI color and cursor escape sequences from terminal output.
pub fn strip_ansi(text: &str) -> String {
    ANSI_REGEX.replace_all(text, "").into_owned()
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_strip_ansi() {
        let input = "\x1b[31mError:\x1b[0m Failed test";
        assert_eq!(strip_ansi(input), "Error: Failed test");
    }
}
